odoo/modules/module.py

```diff
@@ -20,6 +19,21 @@
 import odoo.release as release
 from odoo.tools.misc import file_path
 
+try:
+    from packaging.requirements import InvalidRequirement, Requirement
+except ImportError:
+    class InvalidRequirement(Exception):
+        ...
+
+    class Requirement:
+        def __init__(self, pydep):
+            if not re.fullmatch(r'\w+', pydep):  # check that we have no versions or marker in pydep
+                msg = f"Package `packaging` is required to parse `{pydep}` external dependency and is not installed"
+                raise Exception(msg)
```

Comment:
The goal is that the exception is not catched by the check memthod so that the error message that packaging is missing will reach the user as it is. If an external dep pools like, `SomePackage==5.0.1`, the user will have the complete info that `packaging` is missing before being able to support such a kind of dependency. The error message is raised the same way as if `SomePackage` was missing. If we raise an InvalidRequirement here, it would look the same way as if `SomePackage` is missing, even if it is present.